README.md

```diff
@@ -54,6 +54,39 @@ When adding a new pre-commit check, please run Pre-Commit manually first:
 
     % pre-commit run --all-files
 
+### Pull Request Methodology
+We would like, as much as possible, to maintain a linear git history,
+since it is simple, readable, and easier to bisect in case of bugs.
+We would also like to have pull requests reviewed, and finally merged
+responsibly.
+Therefore, our recommendation to all contributors is to follow the following
+procedure:
+1.  Create a [fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo)
+    of `herdtools7` named, for example, `upstream` and set `herdtools7` as your remote.
+2.  Create a git branch: `git checkout -b feature`
+3.  Implement new feature/change.
+4.  Commit with meaningful messages:
+    `git commit -m "[tag] one line descriptive title, followed by detailed description"`
+    The tag(s) should refer to the component that the feature relates to, e.g.,
+    `[herd]`, `[litmus]`, etc. Generally, the tags correlate to the sub-folders.
+    Also, `[all]` applies to changes that effect several components.
+5.  Test the feature and make sure all relevant tests pass, including
+    `make test`.
+6.  Assuming `origin` is your clone of `herdtools7`, push with
+    `git push --set-upstream origin feature`
+7.  Rebase often, repeating 2--6 as much as needed:
```

Comment:
Strictly speaking not necessary for the sequence of steps here, but I wonder if it is useful to keep origin's master branch up to date with upstream's master branch. Presumably, step 2 is good to do branching off up-to-date master.

I usually press a button on the web interface for this ("Sync fork" on the forked page).